admin/controllers/pupil.js

```diff
@@ -0,0 +1,257 @@
+const moment = require('moment')
+
+const School = require('../models/school')
+const Pupil = require('../models/pupil')
+const errorConverter = require('../lib/error-converter')
+const ValidationError = require('../lib/validation-error')
+const addPupilErrorMessages = require('../lib/errors/pupil').addPupil
+const pupilValidator = require('../lib/validator/pupil-validator')
+const { fetchPupilsData, fetchPupilAnswers } = require('../services/pupilService')
+
+const getAddPupil = async (req, res, next) => {
+  res.locals.pageTitle = 'Add pupil'
+  const schoolId = req.user.School // school object from session
```

Comment:
These are carried changes from before.
Let me know if there's anything else.
Updated in the last commit